### PR TITLE
feat(simplecache): allow specifying custom simplecache path

### PR DESCRIPTION
**Pull request by hypeJunction**
Sites can now define a custom location to store cached views by providing
$CONFIG->assetroot value.

### CODE REVIEW

File: engine/classes/Elgg/Di/ServiceProvider.php [hunk 1]
```diff
@@ -751,12 +751,19 @@ public function initConfig(Config $config, ServiceProvider $sp) {
...
 		} else {
-			$config->cacheroot = $config->dataroot . 'caches/';
+			$config->cacheroot = Paths::sanitize($config->dataroot . 'caches');
 		}
 		$lock('cacheroot');
 
+		if ($config->assetroot) {
+			$config->assetroot = Paths::sanitize($config->assetroot);
+		} else {
+			$config->assetroot = Paths::sanitize('views_simplecache');
```
Comment (jeabakker) [inline engine/classes/Elgg/Di/ServiceProvider.php]: shouldn't this be something like `$config->cachroot . 'views_simplecache'`?

Comment (jeabakker) [reply engine/classes/Elgg/Di/ServiceProvider.php]: I mean in `Paths::sanitize`

Comment (hypeJunction) [reply engine/classes/Elgg/Di/ServiceProvider.php]: Yes, it should. Question for you, do we want to keep these BC? If someone has a symlink setup for the old location, we would probably want to keep it there, no?

Comment (jeabakker) [reply engine/classes/Elgg/Di/ServiceProvider.php]: Or rewrite the symlink to the new location (if possible). If not possible maybe add admin notice.

I think don't try to hard to keep BC, just mention something should manually be done

**Comment by hypeJunction** [comment]
To hell with it. Another endless snowball.

**Pull request closed without merging**